server/datastore/mysql/hosts.go

```diff
@@ -2755,31 +2756,13 @@ func (ds *Datastore) SetOrUpdateMDMData(
 	installedFromDep bool,
 	name string,
 ) error {
-	// MDM queries return an empty server URL when the host is not enrolled to an MDM.
-	if serverURL == "" {
-		// We use the reader even if it might miss some not replicated rows,
-		// because it will eventually catch up on subsequent ingestions and
-		// the entry will be deleted.
-		var id uint
-		switch err := sqlx.GetContext(ctx, ds.reader(ctx), &id,
-			`SELECT host_id FROM host_mdm WHERE host_id = ?`, hostID,
-		); {
-		case err == nil:
-			if _, err := ds.writer(ctx).ExecContext(ctx, `DELETE FROM host_mdm WHERE host_id = ?`, hostID); err != nil {
-				return ctxerr.Wrapf(ctx, err, "delete host_mdm row: %d", hostID)
-			}
-			return nil
-		case errors.Is(err, sql.ErrNoRows):
-			return nil
-		default:
-			return ctxerr.Wrapf(ctx, err, "getting host_mdm row: %d", hostID)
+	var mdmID *uint
+	if serverURL != "" {
```

Comment:
As a note to our future selves, I tried running a live query with `SELECT * FROM mdm` in my machine.

**Results when MDM is ON**

| "host_display_name"     | "access_rights" | "checkin_url" | "dep_capable" | "enrolled" | "has_scep_payload" | "identity_certificate_uuid"            | "install_date"              | "installed_from_dep" | "payload_identifier"          | "server_url"                                   | "sign_message" | "topic"                                                        | "user_approved" |
|-------------------------|-----------------|---------------|---------------|------------|--------------------|----------------------------------------|-----------------------------|----------------------|-------------------------------|------------------------------------------------|----------------|----------------------------------------------------------------|-----------------|
| "Roberto’s MacBook Pro" | "8191"          | ""            | "false"       | "true"     | "true"             | "BCA53F9D-5DD2-494D-98D3-0D0F20FF6BA1" | "2023-07-14 23:27:35 +0000" | "false"              | "com.fleetdm.fleet.mdm.apple" | "https://roperzh-fleet.ngrok.io/mdm/apple/mdm" | "true"         | "com.apple.mgmt.External.bb775651-0719-454f-98bc-2ee32ab43b7e" | "true"          |

**Results when MDM is OFF**

| "host_display_name"     | "access_rights" | "checkin_url" | "dep_capable" | "enrolled" | "has_scep_payload" | "identity_certificate_uuid" | "install_date" | "installed_from_dep" | "payload_identifier" | "server_url" | "sign_message" | "topic" | "user_approved" |
|-------------------------|-----------------|---------------|---------------|------------|--------------------|-----------------------------|----------------|----------------------|----------------------|--------------|----------------|---------|-----------------|
| "Roberto’s MacBook Pro" | ""              | ""            | "false"       | "false"    | ""                 | ""                          | ""             | ""                   | ""                   | ""           | ""             | ""      | ""              |

Reply:
ouch, I swear that in the preview the tables were scrollable!